lib.php

```diff
@@ -0,0 +1,97 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+use core\session\util;
+
+/**
+ * This file contains functions used by tool_heartbeat.
+ *
+ * @package    tool_heartbeat
+ * @copyright  2019 Tom Dickman <tomdickman@catalyst-au.net>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+/**
+ * Return an error that ELB will pick up
+ *
+ * @param string $reason
+ */
+function failed($reason) {
+    // Status for ELB, will cause ELB to remove instance.
+    header("HTTP/1.0 503 Service unavailable: failed $reason check");
+    // Status for the humans.
+    print "Server is DOWN<br>\n";
+    echo "Failed: $reason";
+    exit;
+}
+
+/**
+ * Checks if the command line maintenance mode has been enabled. Skip the config bootstrapping.
+ *
+ * @param string $configfile The relative path for config.php
+ * @return bool True if climaintenance.html is found.
+ */
+function check_climaintenance($configfile) {
```

Comment:
lib.php should be minimal and for callbacks, I think this should be called locallib.php instead

Reply:
moved to locallib.php